do-not-commit/useless-comparison.py

```diff
@@ -0,0 +1,23 @@
+# ruleid:no-strings-as-booleans
+if "detached HEAD" or "master" in expected:
```

Comment:
Using strings as booleans in Python has unexpected results.
`"one" and "two"` will return "two".
`"one" or "two"` will return "one".
 In Python, strings are truthy, evaluating to True. ([.useless-comparison.no-strings-as-booleans](https://semgrep.dev/r/.useless-comparison.no-strings-as-booleans))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=.useless-comparison.no-strings-as-booleans) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>